api/proto/teleport/plugins/v1/plugin_service.proto

```diff
@@ -113,6 +113,18 @@ message GetAvailablePluginTypesResponse {
   repeated PluginType plugin_types = 1;
 }
 
+// GetPluginStaticCredentialsRequest is the request type for GetPluginStaticCredentials
+message GetPluginStaticCredentialsRequest {
```

Comment:
Are the static credentials global/shared for all plugins?

How many credentials are we expected to see in practice?

Reply:
The credentials are not shared for all plugins, and the number of credentials is likely to be 1 per plugin save in the (future) case of rotation which is not yet implemented.

Reply:
OK, so should this be part of Plugin/GetPlugin then?

If not, then how is the plugin specified in the request?

Reply:
It shouldn't because there's a possibility of later rotating them -- it's possible to get many plugin credentials back for a plugin. So there's a 1 to many mapping.

Reply:
But how is the plugin specified? Do we expect the labels to always match a single plugin?

Reply:
Ahhh, I gotcha. So the credentials are created along with the plugin. CreatePluginRequest currently looks like this:

```go
// CreatePluginRequest creates a new plugin from the given spec and initial credentials.
message CreatePluginRequest {
  // Plugin is the plugin object without live credentials.
  types.PluginV1 plugin = 1;
  // BootstrapCredentials are the initial credentials
  // issued by e.g. OAuth2 authorization code flow.
  // In the scope of processing this request, these are exchanged for
  // short-lived renewable credentials, which are stored in the Plugin.
  types.PluginBootstrapCredentialsV1 bootstrap_credentials = 2;
  // Static credentials are optional static credentials
  // to supply to the plugin.
  types.PluginStaticCredentialsV1 static_credentials = 3;
}
```

We do indeed expect the labels to always match a single plugin.

Reply:
My "gut design" would be to nest everything inside types.PluginV1 and control the reads. The underlying storage could split the plugin into separate keys/resources as a way to avoid "accidental" reads.

I see you are following the RFD, so I won't bother you with this any further. Happy to chat design out of band. Thanks for the patient explanations.